Library/Homebrew/formula_installer.rb

```diff
@@ -763,14 +763,23 @@ def build
       raise "Empty installation"
     end
   rescue Exception => e # rubocop:disable Lint/RescueException
-    e.options = display_options(formula) if e.is_a?(BuildError)
+    # Utils.safe_fork rebuilds ErrorDuringExecutions for us, but doesn't have access
+    # to the Formula object that we need to rebuild BuildErrors.
+    if e.is_a?(ChildProcessError) && e.inner["json_class"] == "BuildError"
```

Comment:
Yeah, I could make this do a `respond_to? :inner_class` if we're good on the `inner_class` solution above.

Reply:
Actually: like above I think it might be nice to fiddle with handling `ChildProcessError` elsewhere so this special handling isn't required at all here.